### DISCUSSION TITLE
SPIKE color sesnor keeps report green when see half black line

### DISCUSSION THREAD

**Discussion by shihhaoyang** [pybricks/support#1760] [Q&A] [answered]
Hi everyone,

My students are using Pybricks block code to prepare for RoboCUP line following. We've encountered a problem that I'm struggling to solve. 

The goal is to have the robot follow a line, and when its color sensor detects a green intersection, turn onto that path. However, when we add the code for green intersection detection, the robot keeps turning. This seems to be because the sensor sometimes reports a green color when it sees half of the black line. 

We tried resetting the color HSV values using "my black", "my white", and "my green" from the color sensor readings, but this hasn't fixed the issue.  

Any suggestions on how to resolve this would be greatly appreciated!
attached the test code.
![PXL_20240815_123714576](https://github.com/user-attachments/assets/781999d3-112c-4be4-a525-5fd0c46cca73)

![Screenshot 2024-08-16 003522](https://github.com/user-attachments/assets/d1c138f6-90b3-40d3-bc06-69610b47ab67)

Thanks,
Sam

**Comment by laurensvalk** [comment, marked as answer]
The code you're showing looks good, I think.

When you change that loop to just print the colors, does it give you the correct result? In other words, were both of your pictures above taken with the robot in the same place?

Can you try setting the S value for black and white to 0? And perhaps increase the S value for green a bit. This should make it less likely to report green when it isn't seeing it.

**Reply by shihhaoyang** [reply]
Thanks a lot. Changing the S value to 0 for black and white does fix the problem. Now it works beautifully.
